### PR TITLE
preserving files permission while zipping

### PR DESCRIPTION
**Pull request by tanhe123**
fix https://github.com/aliyun/fun/issues/36

### CODE REVIEW

**Comment by vangie** [comment]
LGTM

File: lib/zip.js [hunk 1]
```diff
@@ -26,6 +27,10 @@ function globAsync(pattern, cwd) {
   });
 }
 
+function permStr(mode) {
+  return (mode & parseInt('777', 8)).toString(8);
```
Comment (JacksonTian) [inline lib/zip.js]: return (mode & 0o777).toString(8);

Comment (JacksonTian) [reply lib/zip.js]: 可以用8进制字面量

Comment (tanhe123) [reply lib/zip.js]: done

**Pull request merged**